contentprovider.go

```diff
@@ -552,6 +552,35 @@ func findSection(secs []DocumentSection, off, sz uint32) (uint32, bool) {
 	return 0, false
 }
 
+// findMaxOverlappingSection returns the index of the section in secs that
+// overlaps the most with the area defined by off and sz. If no section
+// overlaps, it returns 0, false.
+//
+// The implementation assumes that sections do not overlap.
+func findMaxOverlappingSection(secs []DocumentSection, off, sz uint32) (uint32, bool) {
+	// Find the first section that overlaps.
+	j := sort.Search(len(secs), func(i int) bool { return secs[i].End > off })
```

Comment:
do you have an example where it is important to find the max overlap? Can't we just score this based on the first overlap found. If I am not mistaken it doesn't affect the score since any overlap you get a boost.

Reply:
I thought a bit more about this and went the following route: We now calculate the relative overlap and return the section with the largest relative overlap. This way we prefer complete over partial overlaps, even if the partial overlap is bigger. It's important which section we pick because the boost depends on the symbol type behind the section.